lib/components/datepicker/date_picker.dart

```diff
@@ -0,0 +1,62 @@
+import 'package:flutter/material.dart';
+import 'package:intl/intl.dart';
+
+class TUIDatePicker extends StatefulWidget {
+  final String placeholder;
+  final String dateFormat;
+  final DateTime? minimumDate;
+  final DateTime? maximumDate;
+  final Function(DateTime)? dateSelected;
+
+  const TUIDatePicker(
+      {required this.placeholder,
+      this.dateFormat = 'yyyy-MM-dd',
+      this.minimumDate,
+      this.maximumDate,
+      this.dateSelected});
+
+  @override
+  State<TUIDatePicker> createState() => _TUIDatePickerState();
+}
+
+class _TUIDatePickerState extends State<TUIDatePicker> {
+  final TextEditingController _dateController = TextEditingController();
+  DateTime? _selectedDate;
+
+  @override
+  void initState() {
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return TextField(
```

Comment:
Let's use TUITextField so that all the styling it does on the text field, which We get for free here.

Reply:
@kalpeshp0310 We have a `TUIInputField` but we cannot use that because we cannot reset the controller once the instance is created. I already tried that.